flake.nix

```diff
@@ -15,7 +15,40 @@
           inherit system;
           overlays = [ inputs.rust-overlay.overlays.rust-overlay ];
         };
-      in {
+        base-git-cliff = { buildType }: pkgs.rustPlatform.buildRustPackage {
+          name = "git-cliff";
+          src = ./.;
+          cargoLock = {
+            lockFile = ./Cargo.lock;
+          };
+          checkType = "debug";
+          inherit buildType;
+          checkFlags = [
+            "--skip=run_os_command"
+            "--skip=git_tags"
+            "--skip=git_upstream_remote"
+            "--skip=includes_root_commit"
+            "--skip=resolves_existing_tag_with_name_and_message"
+            "--skip=resolves_tag_when_no_tags_exist"
+            "--skip=test_should_retain_commit"
+            "--skip=commit_search"
+            "--skip=get_latest_commit"
+            "--skip=get_latest_tag"
```

Comment:
Why are we skipping these tests?

Reply:
IIUC, they run filesystems/git-based tests.

I can't investigate if you want.

Reply:
Yes, would be nice to run them as a part of the Nix build :)

Reply:
You can't really do that out-of-the-box.

Nix flake drops `.git` directory at build time.

A solution which could be used would be to create a (dynamically or not) a git repository for the test (we could reuse `git-cliff`, but GitHub does not allow you to download and archive with the full history).

WDYT?

Reply:
I see, then in that case let's skip them via `--skip "repo::test"` (instead of individually skipping them) which is the same thing that we do for the [Arch package](https://gitlab.archlinux.org/archlinux/packaging/packages/git-cliff/-/blob/main/PKGBUILD?ref_type=heads#L35).

Reply:
done!

I won't lie, it was challenging.

Reply:
Hmm, I'm a bit confused. Why are we adding a new rust flag instead of just `cargo test -- --skip repo::test --skip command`?

Reply:
ah, I thought you wanted something more fine coarse.

Reply:
This is also fine, but the `--skip` approach would be much simpler I think :)